### ISSUE TITLE
DebugEngine does not pass default value for getMultiple

### ISSUE THREAD

**Issue by xyng** [cakephp/debug_kit#781]
DebugKit wraps Cache-Engines with the `DebugKit\Cache\Engine\DebugEngine` class.
This class reimplements the function `getMultiple` to track performance (i guess).
That function takes two parameters: `$keys` (the keys to get from cache) and `$default` (the value to return when there was no cache hit) and internally calls the `getMultiple` on the real engine ([see here](https://github.com/cakephp/debug_kit/blob/7e0de3676c867318d76c1293ff7fa8fbf075e1fc/src/Cache/Engine/DebugEngine.php#L199)).
However, the param `$default` is not passed to the call to the engine.

This breaks when using CakePHPs `Cache::pool` with `Psr16Adapter` from Symphony - which a library I'm using requires me to (it internally wraps a given pool).
This breaks because said Adapter in Symphony internally uses a reference to `\stdObject` to determine whether the cache had a hit or not (which allows them to store `null`-Values in the Cache). And thus, cache-misses are not recognised and null is returned as the cached value.

Said `Psr16Adapter` requires the given pool to implement `Psr\SimpleCache\CacheInterface`, as which the return-value of CakePHPs `Cache::pool` is defined (as well as the class `Cake\Cache\CacheEngine`, which said `DebugEngine` extends).

Is there some intention why this param is not passed?
If not I'm happy to send a pull-request to fix this issue (if you'd like one).

**Comment by othercorey** [comment]
Can you show what code you're talking about? It's a little hard to understand what you're saying.

**Comment by markstory** [comment]
The default value not being passed is a mistake. A pull request to fix that would be greatly appreciated.

**Comment by xyng** [comment]
@markstory thanks. i'll send one.

@othercorey here's a short example:

```php
use Symfony\Component\Cache\Adapter\Psr16Adapter;
use Cake\Cache\Cache;

// constructor takes Psr\SimpleCache\CacheInterface - which is the return type of Cache::pool
$adapter = new Psr16Adapter(Cache::pool('default));

$item = $adapter->get('test', function () { return 42 });

// $item will always be `null` when DebugKit is enabled, because the default value is not forwarded.
// `null` is the fallback-value for default - but something else is given by the wrapper. since the returned value does not equal the given default value, the returned value (-> `null`) is expected to be the value from the cache
// because of this, the callback that would generate the cache-value is never called
```

**Comment by othercorey** [comment]
Fix released in 4.3.5.
